cmd/access/main.go

```diff
@@ -341,23 +342,35 @@ func main() {
 
 			return followerEng, nil
 		}).
+		Component("finalized snapshot", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			finalizedSnapshot, err = synceng.NewFinalizedSnapshotCache(node.Logger, node.State, filter.And(
+				filter.HasRole(flow.RoleConsensus),
```

Comment:
My comment is not about this specific line. 

For staked/unstaked sync requests handling, this is what we want:
1. Allow staked nodes to send sync requests to only consensus nodes.
2. Allow consensus nodes to respond sync requests from staked nodes.
3. Allow consensus nodes to `reject` sync requests from unstaked nodes.
4. Allow staked access nodes to respond sync requests from unstaked nodes.

I think `3` should not happen, because unstaked nodes are in a different network than staked  consensus nodes. 

Do we allow non-consensus nodes to respond sync requests? 
For instance, verification nodes also have sync engine, which is able to respond sync requests, but since all staked nodes are sending sync requests to only consensus nodes, no one is actually syncing blocks from verification nodes.
Would it be an optimization to allow verification nodes to sync with each other. 

Do we want the unstaked nodes to sync with each other in order to reduce the load to staked access nodes?
If not, then if we use the new sync engine in unstaked nodes, then they will be able to respond to sync requests, but actually no one is sending sync requests to unstaked nodes, as they all sync with staked access nodes.

Reply:
> Do we allow non-consensus nodes to respond sync requests?
For instance, verification nodes also have sync engine, which is able to respond sync requests, but since all staked nodes are sending sync requests to only consensus nodes, no one is actually syncing blocks from verification nodes even if they are capable to handle.
Would it be an optimization to allow verification nodes to sync with each other?

I believe so, and this is something I've brought up with @vishalchangrani. I think eventually, we would like to be able to have verification node sync with each other, as well as have unstaked nodes sync with each other as well since this will be more scalable.

There is still a little bit of work to be done on the networking side to setup DHT, but once that is ready then we should be able to do this.